tools/library/filter/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+message( STATUS "---------Beginning cmake build of ld-filter---------")
```

Comment:
I would build all of `tools/library` as a single llibrary, and it should be an internal static library, not an installed shared library. This is for two reasons - we don't want to have to version and maintain a shared library interface, and there's lots of performance-critical stuff in the library that can benefit from LTO.